src/gasPriceOracle/oracle.ts

```diff
@@ -47,30 +49,21 @@ async function getEthersGasPriceEstimate(
   chainId?: number,
   legacyFallback = true
 ): Promise<GasPriceEstimate> {
-  if (chainId === undefined) {
-    ({ chainId } = await provider.getNetwork());
-  }
+  chainId ?? ({ chainId } = await provider.getNetwork());
 
   const gasPriceFeeds = {
     [CHAIN_IDs.ALEPH_ZERO]: arbitrum.eip1559,
     [CHAIN_IDs.ARBITRUM]: arbitrum.eip1559,
-    [CHAIN_IDs.BASE]: ethereum.eip1559,
     [CHAIN_IDs.LINEA]: linea.eip1559, // @todo: Support linea_estimateGas in adapter.
     [CHAIN_IDs.MAINNET]: ethereum.eip1559,
-    [CHAIN_IDs.MODE]: ethereum.eip1559,
-    [CHAIN_IDs.OPTIMISM]: ethereum.eip1559,
     [CHAIN_IDs.POLYGON]: polygon.gasStation,
-    [CHAIN_IDs.ZK_SYNC]: ethereum.legacy,
     [CHAIN_IDs.SCROLL]: ethereum.legacy,
+    [CHAIN_IDs.ZK_SYNC]: ethereum.legacy,
   } as const;
 
   let gasPriceFeed = gasPriceFeeds[chainId];
-  if (gasPriceFeed === undefined) {
-    if (!legacyFallback) {
-      throw new Error(`No suitable gas price oracle for Chain ID ${chainId}`);
-    }
-    gasPriceFeed = chainIsOPStack(chainId) ? ethereum.eip1559 : ethereum.legacy;
-  }
+  assert(gasPriceFeed || legacyFallback, `No suitable gas price oracle for Chain ID ${chainId}`);
+  gasPriceFeed ??= chainIsOPStack(chainId) ? op.legacy : ethereum.legacy;
```

Comment:
No 1559 for op stack?

Reply:
In the gasPriceOracle, `legacy` vs `eip1559` really just informs whether the underlying adapter calls ethers `getGasPrice()` or `getFeeData()`.

`getGasPrice()` maps directly to `eth_gasPrice()`, whereas `getFeeData()` pulls the base fee from the latest block and then adds the hardcoded 1.5 Gwei.

So...maybe we'd just be better off to call eth_gasPrice() directly and `eth_getMaxPriorityFeePerGas()` in parallel. I think this would probably solve most of our issues.

btw, for OP stack, `eth_getMaxPriorityFeePerGas()` just returns 0xf4240 (0.001 Gwei).

Reply:
That seems better than the 1.5 gwei default!

> btw, for OP stack, eth_getMaxPriorityFeePerGas() just returns 0xf4240 (0.001 Gwei).